src/query/cypher_query_interpreter.cpp

```diff
@@ -125,19 +125,20 @@ ParsedQuery ParseQuery(const std::string &query_string, UserParameters const &us
 }
 
 std::unique_ptr<LogicalPlan> MakeLogicalPlan(AstStorage ast_storage, CypherQuery *query, const Parameters &parameters,
-                                             DbAccessor *db_accessor,
+                                             DbAccessor *db_accessor, DatabaseAccessProtector db_acc,
```

Comment:
Can we pass in the protector at run-time instead of at plan creation time?
If we cache the plan with a protector, we need to make sure the cache is destroyed before attempting to delete the storage. Otherwise we will be blocked.

Reply:
yes, I will pass it inside ExecutionContext, I didn't think of this beforehand